src/main/java/net/openhft/chronicle/bytes/internal/CommonMappedBytes.java

```diff
@@ -591,6 +591,12 @@ public String toString() {
                 '}';
     }
 
+    @Override
+    // Avoid reading contents as it may be empty or point to unmapped region.
+    public int hashCode() {
+        return mappedFile.hashCode();
```

Comment:
This is a fantastic change - I had not realised that the reference accounting called hashCode()

Reply:
This is the same problem as for ByteBuffer. It is questionable to have hashCode() and equals() implementations like these for memory regions. Make sure equals() is overridden in the same way as hashCode() !

Reply:
I have added equals() @minborg